integrations/google_ai/tests/generators/chat/test_chat_gemini.py

```diff
@@ -198,6 +225,39 @@ def get_current_weather(location: str, unit: str = "celsius"):  # noqa: ARG001
     assert len(res["replies"]) > 0
 
 
+@pytest.mark.skipif(not os.environ.get("GOOGLE_API_KEY", None), reason="GOOGLE_API_KEY env var not set")
+def test_run_with_streaming_callback():
+    streaming_callback_called = False
+
+    def streaming_callback(_chunk: StreamingChunk) -> None:
+        nonlocal streaming_callback_called
+        streaming_callback_called = True
+
+    def get_current_weather(location: str, unit: str = "celsius"):  # noqa: ARG001
+        return {"weather": "sunny", "temperature": 21.8, "unit": unit}
+
+    get_current_weather_func = FunctionDeclaration.from_function(
+        get_current_weather,
+        descriptions={
+            "location": "The city and state, e.g. San Francisco, CA",
+            "unit": "The temperature unit of measurement, e.g. celsius or fahrenheit",
+        },
+    )
+
+    tool = Tool(function_declarations=[get_current_weather_func])
+    gemini_chat = GoogleAIGeminiChatGenerator(model="gemini-pro", tools=[tool], streaming_callback=streaming_callback)
+    messages = [ChatMessage.from_user(content="What is the temperature in celsius in Berlin?")]
+    res = gemini_chat.run(messages=messages)
+    assert len(res["replies"]) > 0
+
+    weather = get_current_weather(res["replies"][0].content)
+    messages += res["replies"] + [ChatMessage.from_function(content=weather, name="get_current_weather")]
+
+    res = gemini_chat.run(messages=messages)
+    assert len(res["replies"]) > 0
```

Comment:
```suggestion
```

If we want to just test the streaming feature, this part can probably be removed from this test